formstack-consents/src/main/scala/com/gu/identity/formstackconsents/IdentityClient.scala

```diff
@@ -0,0 +1,66 @@
+package com.gu.identity.formstackconsents
+
+import io.circe.syntax._
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyResponseEvent
+import com.gu.identity.formstackconsents.Lambda.{FormstackSubmission, Config}
+import com.typesafe.scalalogging.StrictLogging
+import io.circe.generic.extras._
+import scalaj.http.{Http, HttpResponse}
+import scala.util.Try
+
+class IdentityClient(config: Config) extends StrictLogging {
+
+  val newsletters: List[Newsletter] = List(Holidays, Students, Universities, Teachers, Masterclasses, SocietyWeekly, EdinburghFestivalDataCollection)
+
+  def getResponseFromIdentity(formstackSubmission: FormstackSubmission, newsletter: Newsletter): Option[HttpResponse[String]] = {
+    // The key in the JSON sent to Identity depends on the listType. Sometimes the listType is 'set-lists' with a value of the consent name,
+    // and sometimes the listType is 'set-consents'. See example below.
+    //  {
+    //    "email" : "lauren.emms@guardian.co.uk",
+    //    "set-consents" : "holidays"
+    //  }
+
+    implicit val circeConfig: Configuration = Configuration.default.copy(
```

Comment:
Interesting approach. Ideally, you don't want to tie down the `IdentityRequest` class to a specific class instance, however, I can see why it's necessary if you're using the `@ConfiguredJsonCodec` annotation. Another approach could be to define `IdentityRequest` statically somewhere (e.g. the companion object) along with a function: `Newsletter => Encoder[IdentityRequest]` which could then be called within this method to define an implicit encoder for the given news letter.

Reply:
I have created a companion object with a method in, it needs the implicit val set within the same scope as defining the case class. Is this what you meant by the Newsletter => Encoder[IdentityRequest] function?